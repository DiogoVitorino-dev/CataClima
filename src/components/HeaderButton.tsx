import {MaterialIcons} from '@expo/vector-icons';
import {Link} from 'expo-router';
import {Pressable, StyleProp, TextStyle} from 'react-native';

export default function HeaderButton({
  href,
  params,
  onPress,
  icon,
  iconSize,
  iconColor,
  style,
}: {
  href?: string;
  params?: object;
  onPress?:() => void;
  iconSize: number;
  icon: string | any;
  iconColor: string;
  style?: StyleProp<TextStyle>;
}) {
  return (
    <Link 
      href={{pathname: href, params: params}} 
      style={style} 
      asChild
    >
      <Pressable 
        accessibilityRole='button'
        onPress={() => onPress ? onPress() : null}>
        {({pressed}) => (
          <MaterialIcons 
            name={icon} 
            size={iconSize} 
            color={iconColor} 
            style={{opacity: pressed ? 0.5 : 1}} 
          />
        )}
      </Pressable>
    </Link>
  );
}
