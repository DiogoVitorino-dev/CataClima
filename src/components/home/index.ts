export * from './AnimatedBackground';
export * from './WeatherText';
export * from './WeatherStatus';
export * from './WeatherNoData';
export * from './WeatherItem';
export * from './WeatherDetail';