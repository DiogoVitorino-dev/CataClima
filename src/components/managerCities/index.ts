export * from './CitiesList';
export * from './UserCitiesList';
export * from './ListItem';