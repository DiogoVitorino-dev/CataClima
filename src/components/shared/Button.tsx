import {Pressable, StyleProp, StyleSheet, ViewStyle, useColorScheme} from 'react-native';
import { View } from './Themed';
import { Colors } from '../../constants';
import { OpenTextStyled } from './StyledText';

export function Button({
	label,
	disabled,
	onPress,
	style,
}: {
  label: string;  
  disabled?: boolean;
  onPress: () => void;
  style?: StyleProp<ViewStyle>;
}) {
	const colorScheme = useColorScheme();

	return (
    
		<View 
			style={[
				styles.container,
				style,{
					borderColor:Colors[colorScheme ?? 'light'].borderColor,
					borderWidth:StyleSheet.hairlineWidth
				}
			]}
			lightColor="#FAFAFA"
			darkColor="#1A1A1A">
			<Pressable 
				style={styles.button}
				testID='pressableTestID'
				onPress={() => onPress()} disabled={disabled}>          
				{({pressed}) => (
					<OpenTextStyled 
						style={[styles.label, {opacity: pressed ? 0.5 : 1}]}>
						{label}
					</OpenTextStyled>
				)}
			</Pressable>
		</View>
	);
}

const styles = StyleSheet.create({
	container: {
		maxHeight: 40,
		justifyContent: 'center',
		alignItems: 'center',
		borderRadius: 20,
		shadowColor: 'rgba(0,0,0,0.3)',
		shadowOffset: {width: 5, height: 5},
		shadowRadius: 10,
		elevation: 5,
	},

	button: {
		width: '100%',
		height: '100%',
		paddingHorizontal: 20,
		paddingVertical: 5,
		justifyContent: 'center',
		alignItems: 'center',
	},

	label: {
		textAlign: 'center',
		fontSize: 16,
	},
});
