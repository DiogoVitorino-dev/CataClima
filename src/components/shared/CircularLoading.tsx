import { ActivityIndicator, StyleProp, ViewStyle } from "react-native";

import { useAppColor } from "@/hooks/ThemeHooks";

interface IProps {
  visible?: boolean;
  size?: number | "small" | "large";
  color?: string;
  style?: StyleProp<ViewStyle>;
}

export function CircularLoading({ color, visible, size, style }: IProps) {
  const appColor = useAppColor();
  return (
    <ActivityIndicator
      style={style}
      animating={visible || false}
      size={size}
      color={color || appColor.tint}
    />
  );
}
