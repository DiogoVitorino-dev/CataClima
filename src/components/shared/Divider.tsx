import React from "react";
import { DimensionValue, View } from "react-native";

import { useAppColor } from "@/hooks/ThemeHooks";

interface IProps {
  color?: string;
  width?: DimensionValue;
  height?: number;
  lineSize?: DimensionValue;
}

export default function Divider({ color, height, lineSize, width }: IProps) {
  const appColor = useAppColor();
  return (
    <View
      style={{
        backgroundColor: color || appColor.borderColor,
        height: lineSize || 1,
        width: width || "100%",
        marginVertical: height || 3,
        borderRadius: 3,
      }}
    />
  );
}
