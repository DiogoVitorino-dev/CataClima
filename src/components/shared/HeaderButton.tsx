import { FontAwesome } from "@expo/vector-icons";
import { Pressable, StyleProp, TextStyle } from "react-native";

import { IconsFontAwesome } from "@/constants/TypesApp";

interface IProps {
  onPress: () => void;
  icon: IconsFontAwesome;
  color?: string;
  style?: StyleProp<TextStyle>;
}

export default function HeaderButton({ onPress, icon, color, style }: IProps) {
  return (
    <Pressable accessibilityRole="button" style={style} onPress={onPress}>
      {({ pressed }) => (
        <FontAwesome
          name={icon}
          size={30}
          color={color}
          style={{ opacity: pressed ? 0.5 : 1 }}
        />
      )}
    </Pressable>
  );
}
