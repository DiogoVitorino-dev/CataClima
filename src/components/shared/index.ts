export * from './Button';
export * from './HeaderButton';
export * from './PermissionModal';
export * from './Searchbar';
export * from './SimpleModal';
export * from './Loading';
export * from './StyledText';
export * from './Themed';
export * from './TimeAgo';
