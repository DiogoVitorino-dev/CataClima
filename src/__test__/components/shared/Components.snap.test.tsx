import renderer from 'react-test-renderer'
import { Button, HeaderButton, Loading, PermissionModal, Searchbar, SimpleModal, TimeAgo } from '../../../components/shared';


describe('Components Snapshots', () => {
	it('should Button component renders correctly', () => {    
		const tree = renderer.create(<Button label='test' onPress={()=>{}} />).toJSON();
		expect(tree).toMatchSnapshot()    
	})

	it('should HeaderButton component renders correctly', () => {    
		const tree = renderer.create(
		<HeaderButton
			href='/'
			onPress={()=>{}}
			params={{test:'test'}}
			style={{margin:5}}
			icon='cloud' 
			iconColor='#fff' 
			iconSize={20} 
		/>).toJSON();
		
		expect(tree).toMatchSnapshot()    
	})

	it('should Loading component renders correctly', () => {    
		const tree = renderer.create(<Loading />).toJSON();
		expect(tree).toMatchSnapshot()    
	})

	it('should PermissionModal component renders correctly', () => {    
		const tree = renderer.create(
		<PermissionModal 
		visible 
		onPressFirstButton={()=>{}} 
		onPressSecondButton={()=>{}} 
		onDismiss={()=>{}} />).toJSON();
		expect(tree).toMatchSnapshot()    
	})

	it('should Searchbar component renders correctly', () => {    
		const tree = renderer.create(
		<Searchbar      
			onChangeText={()=>{}}
			placeholder='a'      
			backgroundColor='#fff'
		/>).toJSON();
		expect(tree).toMatchSnapshot()    
	})

	it('should SimpleModal component renders correctly', () => {    
		const tree = renderer.create(<SimpleModal message='test' visible />).toJSON();
		expect(tree).toMatchSnapshot()    
	})

	it('should TimeAgo component renders correctly', () => {
		const date = new Date().toISOString()
		const tree = renderer.create(
		<TimeAgo dateIsoFormat={date} />).toJSON();
		expect(tree).toMatchSnapshot()    
	})
})