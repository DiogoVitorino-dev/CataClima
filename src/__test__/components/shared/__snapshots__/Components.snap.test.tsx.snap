// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Snapshots should Button component renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FAFAFA",
      },
      [
        {
          "alignItems": "center",
          "borderRadius": 20,
          "elevation": 5,
          "justifyContent": "center",
          "maxHeight": 40,
          "shadowColor": "rgba(0,0,0,0.3)",
          "shadowOffset": {
            "height": 5,
            "width": 5,
          },
          "shadowRadius": 10,
        },
        undefined,
        {
          "borderColor": "rgba(0,0,0,0.3)",
          "borderWidth": 0.5,
        },
      ],
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "paddingVertical": 5,
        "width": "100%",
      }
    }
    testID="pressableTestID"
  >
    <Text
      style={
        [
          {
            "color": "#000",
          },
          [
            [
              {
                "fontSize": 16,
                "textAlign": "center",
              },
              {
                "opacity": 1,
              },
            ],
            {
              "fontFamily": "OpenSans",
            },
          ],
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;

exports[`Components Snapshots should HeaderButton component renders correctly 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  href="/?test=test"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "margin": 5,
    }
  }
>
  <Text />
</View>
`;

exports[`Components Snapshots should Loading component renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      {
        "height": 300,
        "width": 300,
      }
    }
  />
</View>
`;

exports[`Components Snapshots should PermissionModal component renders correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onDismiss={[Function]}
  transparent={true}
  visible={false}
/>
`;

exports[`Components Snapshots should Searchbar component renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      [
        {
          "alignItems": "center",
          "borderRadius": 20,
          "flex": 1,
          "flexDirection": "row",
          "marginBottom": 5,
          "maxHeight": 40,
          "maxWidth": 500,
          "paddingHorizontal": 5,
          "width": "100%",
        },
        {
          "backgroundColor": "#fff",
          "borderColor": "rgba(0,0,0,0.3)",
          "borderWidth": 1,
        },
      ],
    ]
  }
>
  <Text />
  <TextInput
    cursorColor="#000"
    inputMode="text"
    onChangeText={[Function]}
    placeholder="a"
    placeholderTextColor="#7D7D7D"
    style={
      [
        {
          "width": "100%",
        },
        {
          "color": "#000",
        },
      ]
    }
  />
</View>
`;

exports[`Components Snapshots should SimpleModal component renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
/>
`;

exports[`Components Snapshots should TimeAgo component renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#fff",
        "fontSize": 14,
        "opacity": 0.8,
      },
      {
        "fontFamily": "OpenSans",
      },
    ]
  }
>
  Há poucos segundos atrás
</Text>
`;
